common/processes/platform/tiny_process_library/tiny_process_library.cpp

```diff
@@ -178,7 +178,33 @@ error::Error Process::AsyncWait(events::EventLoop &loop, AsyncWaitHandler handle
 	return error::NoError;
 }
 
+error::Error Process::AsyncWait(
+	events::EventLoop &loop, AsyncWaitHandler handler, chrono::nanoseconds timeout) {
+	timeout_timer_.reset(new events::Timer(loop));
+
+	auto err = AsyncWait(loop, handler);
+	if (err != error::NoError) {
+		return err;
+	}
+
+	timeout_timer_->AsyncWait(timeout, [this, handler](error::Error err) {
+		// Move timer here so it gets destroyed after this handler.
+		auto timer = std::move(timeout_timer_);
```

Comment:
Q: Are we always _sure_ that it gets moved?

Reply:
Yes, `unique_ptr` doesn't allow copying at all.